board-automation.yml fails on personal (user-owned) accounts: org+user GraphQL query trips set -e

## Bug: `board-automation.yml` fails on personal (user-owned) accounts

**Scaffold version:** project-init 0.1.0, `obsidian-only` preset, `language: python`.

### Symptom
On a repo owned by a **personal account** (not an organization), the `Board automation` workflow's `Sync project item fields` step fails:

```
gh: Could not resolve to an Organization with the login of '<user>'.
##[error]Process completed with exit code 1.
```

This happens on every issue/PR event once `PROJECT_TOKEN` is configured, so the board never updates. (Note: this is only reached after `PROJECT_TOKEN` is a classic PAT with `project` scope — fine-grained PATs separately fail to *write* to user-owned Projects v2 with "Resource not accessible by personal access token".)

### Root cause
The project-lookup query in `.github/workflows/board-automation.yml` requests **both** the `user(login:$owner)` and `organization(login:$owner)` project paths in a single GraphQL call:

```graphql
query($owner: String!, $number: Int!) {
  user(login: $owner)         { projectV2(number: $number) { ... } }
  organization(login: $owner) { projectV2(number: $number) { ... } }
}
```

For a personal account, the `organization` path returns a GraphQL error ("Could not resolve to an Organization"). `gh api graphql` exits **non-zero** whenever the response contains an `errors` array — even though the response still carries valid `data.user.projectV2`. Because the step runs under `set -euo pipefail`, the non-zero exit kills the step at the `PROJECT_DATA=$(...)` assignment, before the existing `jq '.data.user.projectV2 // .data.organization.projectV2 // empty'` fallback (line ~129) ever runs. (Symmetrically, an org-owned repo would error on the `user` path.)

### Suggested fix
Tolerate the partial-error response so the `jq` fallback can select whichever path applies:

```bash
PROJECT_DATA=$(gh api graphql -f query='...' -f owner="$OWNER" -F number="$PROJECT_NUMBER" 2>/dev/null || true)
```

The downstream `jq // empty` + `PROJECT_ID` empty-check already handle the "neither path matched" case gracefully. Alternatively, split into two separate queries, or detect the owner type first.

---
**Related defects found on the same fresh scaffold (v0.1.0):** #208 (CI Python version / matrix), #209 (`uv sync --extra dev` vs `[dependency-groups]`), #210 (`validate-pr.yml` missing `GH_REPO`).

--- a/templates/base/dot_github/workflows/board-automation.yml
+++ b/templates/base/dot_github/workflows/board-automation.yml
@@ -118,9 +118,15 @@ jobs:
                   }
                 }
               }
-            }' -f owner="$OWNER" -F number="$PROJECT_NUMBER")
+            }' -f owner="$OWNER" -F number="$PROJECT_NUMBER" 2>/dev/null || true)
 
-          PROJECT=$(echo "$PROJECT_DATA" | jq -c '.data.user.projectV2 // .data.organization.projectV2 // empty')
+          # A personal (user-owned) account errors on the organization(...) path
+          # (and an org errors on the user(...) path); gh exits non-zero whenever
+          # the response carries an `errors` array, even though valid data.user /
+          # data.organization is still on stdout. Tolerate that so the jq fallback
+          # below can select whichever path applies. Total failures leave
+          # PROJECT_DATA empty -> PROJECT_ID empty -> graceful skip. (PI-207)
+          PROJECT=$(echo "$PROJECT_DATA" | jq -c '.data.user.projectV2 // .data.organization.projectV2 // empty' 2>/dev/null || true)
           PROJECT_ID=$(echo "$PROJECT" | jq -r '.id // empty')
 
           if [ -z "$PROJECT_ID" ]; then

--- a/templates/base/dot_github/workflows/board-automation.yml
+++ b/templates/base/dot_github/workflows/board-automation.yml
@@ -24,6 +24,10 @@ jobs:
     runs-on: ubuntu-24.04
     env:
       GH_TOKEN: ${{ secrets.PROJECT_TOKEN }}
+      # No actions/checkout in this job (it only needs API access); pin the repo
+      # so every `gh` call (e.g. `gh issue view`) works checkout-free instead of
+      # failing on a missing git remote (PI-234).
+      GH_REPO: ${{ github.repository }}
       OWNER: ${{ github.repository_owner }}
       REPO_NAME: ${{ github.event.repository.name }}
       PROJECT_NUMBER: 1

--- a/tests/contracts/test_templates.py
+++ b/tests/contracts/test_templates.py
@@ -128,6 +128,12 @@ class TestScaffoldGitHubFiles:
     def test_board_automation_workflow_created(self):
         assert (self.target / ".github" / "workflows" / "board-automation.yml").is_file()
 
+    def test_board_automation_tolerates_personal_accounts(self):
+        """PI-207: the user+organization project query errors on one path; the
+        lookup must tolerate that so the jq fallback selects whichever applies."""
+        content = (self.target / ".github" / "workflows" / "board-automation.yml").read_text()
+        assert "2>/dev/null || true)" in content
+
     def test_review_status_workflow_created(self):
         assert (self.target / ".github" / "workflows" / "review-status.yml").is_file()
 

--- a/tests/contracts/test_templates.py
+++ b/tests/contracts/test_templates.py
@@ -129,10 +129,12 @@ class TestScaffoldGitHubFiles:
         assert (self.target / ".github" / "workflows" / "board-automation.yml").is_file()
 
     def test_board_automation_tolerates_personal_accounts(self):
-        """PI-207: the user+organization project query errors on one path; the
-        lookup must tolerate that so the jq fallback selects whichever applies."""
+        """PI-207/PI-234: the user+organization project query errors on one path
+        (tolerate it so the jq fallback selects whichever applies), and the
+        checkout-free job must pin the repo so gh calls work without a remote."""
         content = (self.target / ".github" / "workflows" / "board-automation.yml").read_text()
         assert "2>/dev/null || true)" in content
+        assert "GH_REPO:" in content
 
     def test_review_status_workflow_created(self):
         assert (self.target / ".github" / "workflows" / "review-status.yml").is_file()

--- a/tests/contracts/test_templates.py
+++ b/tests/contracts/test_templates.py
@@ -133,7 +133,26 @@ class TestScaffoldGitHubFiles:
         (tolerate it so the jq fallback selects whichever applies), and the
         checkout-free job must pin the repo so gh calls work without a remote."""
         content = (self.target / ".github" / "workflows" / "board-automation.yml").read_text()
-        assert "2>/dev/null || true)" in content
+        lines = content.splitlines()
+        # Target the specific PROJECT_DATA assignment instead of matching the
+        # error-suppression substring anywhere in the file (brittle to harmless
+        # whitespace/formatting changes): locate the multi-line `gh api graphql`
+        # command substitution and capture it through the line that closes it.
+        starts = [
+            i for i, line in enumerate(lines) if line.lstrip().startswith("PROJECT_DATA=")
+        ]
+        assert starts, "board-automation.yml must assign PROJECT_DATA from a gh query"
+        start = starts[0]
+        ends = [i for i, line in enumerate(lines[start:], start) if "|| true" in line]
+        assert ends, "the PROJECT_DATA assignment must close with an error-tolerant guard"
+        assignment = "\n".join(lines[start : ends[0] + 1])
+        # The query must hit the Projects API and tolerate the errors path a
+        # personal (user-owned) account produces on the organization(...) branch,
+        # so the jq fallback can still select whichever path applies. (PI-207)
+        assert "gh api graphql" in assignment
+        assert "projectV2" in assignment
+        assert "2>/dev/null" in assignment
+        assert "|| true" in assignment
         assert "GH_REPO:" in content
 
     def test_review_status_workflow_created(self):

